packages/block-library/src/rss/index.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * Server-side rendering of the `core/rss` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the rss error message.
+ *
+ * @param string $message RSS error message.
+ * @return string|void Void if renders on frontend.
+ */
+function render_rss_error_message( $message ) {
+	return '<div class="components-placeholder"><div class="notice notice-alt notice-error">' . $message . '</div></div>';
+}
+
+/**
+ * Renders the `core/rss` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the block content with received rss items.
+ */
+function render_block_core_rss( $attributes ) {
+
+	$url = $attributes['feedURL'];
+
+	if ( empty( $url ) ) {
+		return;
+	}
+
+	while ( stristr( $url, 'http' ) !== $url ) {
```

Comment:
What does this loop do? Can you explain? It might be worth adding code comment. I see the same code in core 🤷‍♂️

Reply:
It trims the string until it starts with `http`. I don't think we need this here anymore, because we have other checks in PHP now.

Reply:
Why is this being done? Doesn‘t really make sense to me. If the URL is invalid after just trimming, it‘s invalid.